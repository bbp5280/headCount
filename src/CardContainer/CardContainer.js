import React from 'react';
import Card from '../Card/Card';
import styles from '../index.css';
import PropTypes from 'prop-types';

const CardContainer = ({ allData, compare, toggleStateActive }) => {
  const cards = allData.map(district => {
    return <Card
      location={district.location}
      allData={district.data}
      key={district.location}
      compare={compare}
      toggleStateActive={toggleStateActive} />;
  });

  return (
    <div className="card-container">
      <h2 className="data-info">Percent of Students in Enrolled in Full Day Kinder</h2>
      {cards}
    </div>
  );
};

CardContainer.propTypes = {
  allData: PropTypes.array.isRequired,
  compare: PropTypes.func.isRequired,
  toggleStateActive: PropTypes.func
};

export default CardContainer;
