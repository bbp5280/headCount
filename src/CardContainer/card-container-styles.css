.card-container {
  background-color: #f3f3f3;
  padding-bottom: 7.5%; }

.data-info {
  padding-top: 25px;
  text-align: center; }

hr {
  width: 57%;
  height: 2px;
  border-color: #e64f40;
  margin-left: auto;
  margin-right: auto;
  margin-top: 50px; }
