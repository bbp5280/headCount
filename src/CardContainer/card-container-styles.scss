.card-container {
  background-color: #f3f3f3;
  padding-bottom: 20%;
}

.data-info {
  margin: 25px;
}
