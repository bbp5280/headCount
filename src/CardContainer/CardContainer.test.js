import React from 'react';
import CardContainer from './CardContainer';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import config from '../../test/setup.js';

describe('CARD CONTAINER', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(<CardContainer {... {location:'COLORADO',
                                              allData:[{2004: 0.24}],
                                              key:'COLORADO',
                                              compare:mkFun}}/>);

  it('should match the CardContainer snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
