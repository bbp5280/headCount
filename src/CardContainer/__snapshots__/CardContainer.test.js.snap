exports[`CARD CONTAINER should match the CardContainer snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    allData={
        Array [
            Object {
              "2004": 0.24,
            },
          ]
    }
    compare={[Function]}
    location="COLORADO" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="data-info">
          Percent of Students in Enrolled in Full Day Kinder
</h2>,
        Array [
          <Card
            allData={undefined}
            compare={[Function]}
            location={undefined} />,
        ],
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Percent of Students in Enrolled in Full Day Kinder",
          "className": "data-info",
        },
        "ref": null,
        "rendered": "Percent of Students in Enrolled in Full Day Kinder",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "allData": undefined,
          "compare": [Function],
          "location": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="data-info">
            Percent of Students in Enrolled in Full Day Kinder
</h2>,
          Array [
            <Card
              allData={undefined}
              compare={[Function]}
              location={undefined} />,
          ],
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Percent of Students in Enrolled in Full Day Kinder",
            "className": "data-info",
          },
          "ref": null,
          "rendered": "Percent of Students in Enrolled in Full Day Kinder",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "allData": undefined,
            "compare": [Function],
            "location": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
